src/config/mongoose_config_validator_toml.erl

```diff
@@ -323,3 +345,6 @@ validate_password_format(Value) -> validate_enum(Value, [scram, plain]).
 
 validate_pool_scope(Value) when is_binary(Value) -> validate_non_empty_binary(Value);
 validate_pool_scope(Value) -> validate_enum(Value, [host, global]).
+
+validate_root_or_host_config([]) -> ok;
+validate_root_or_host_config([{host, _}, <<"host_config">>]) -> ok.
```

Comment:
I understand that this extra check should apply only to those paths which allow to be set either global and per-host, right? In that case, why doesn't it apply to other `s2s` paths like `[<<"ip_address">>, item, <<"address">>, <<"s2s">>]`? I'm probably missing something about `s2s` configuration, I'm sure 🤔

Reply:
Yes, this is correct, it only applies to the options that are constructed with the `?HOST` macro, creating a function which returns the config for a host. And there are only three s2s options like this: `default_policy`, `shared`, `max_retry_delay`. The `address` one you mentioned contains `host`, but it is not a per-host setting. That `host` is just a regular hostname of an external server.

Reply:
does it mean I need to add `validate_root_or_host_config` for **every** module option validator?

Reply:
@arcusfelis for modules I'd recommend:

```validate([_, <<"modules">>|Path],
         [Value]) ->
    validate_root_or_host_config(Path);```

And then you don't need it for **every** module option validator.